### ISSUE TITLE
[BUG] `namespace` does not allow to export `declaration-merging`

### ISSUE THREAD

**Issue by ttttttssssss** [import-js/eslint-plugin-import#2320]
> reference: [declaration-merging document](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)

### Fail

```ts

namespace P{
  // error: Multiple exports of name 'f'. eslint(import/export)
  export function f () {
  }
  // error: Multiple exports of name 'f'. eslint(import/export)
  export namespace f {}
}


```

**Comment by BrightLin** [comment]
The same.
see https://github.com/typescript-eslint/typescript-eslint/issues/4248
